runtime/include/QuantumDevice.hpp

```diff
@@ -148,17 +148,53 @@ struct QuantumDevice {
      * @param inverse Indicates whether to use inverse of gate
      */
     virtual void NamedOperation(const std::string &name, const std::vector<double> &params,
-                                const std::vector<QubitIdType> &wires, bool inverse) = 0;
+                                const std::vector<QubitIdType> &wires, bool inverse)
+    {
+        NamedOperation(name, params, wires, inverse, {}, {});
+    }
+
+    /**
+     * @brief Apply a single gate to the state vector of a device with its name if this is
+     * supported.
+     *
+     * @param name The name of the gate to apply
+     * @param params Optional parameter list for parametric gates
+     * @param wires Wires to apply gate to
+     * @param inverse Indicates whether to use inverse of gate
+     * @param controlled_wires Controlled wires applied to the operation
+     * @param controlled_values Controlled values applied to the operation
+     */
+    virtual void NamedOperation(const std::string &name, const std::vector<double> &params,
```

Comment:
Why not using the `Modifiers` data type here in the signature of `NamedOperation` and `MatrixOperation`?

```
    virtual void NamedOperation(name, params, wires, modifiers) = 0;
```

Reply:
I see no particular arguments towards or against this, but I would like to limit passing C-pointers around the code. Let's keep it as-is for now, but I am ready to re-design this in the subsequent PR changing the API.